packages/interchain_queries/src/sudo.rs

```diff
@@ -80,8 +106,17 @@ pub fn sudo_tx_query_result(
 
                 // Parse a Send message and check that it has the required recipient.
                 let transfer_msg: MsgSend = MsgSend::decode(message.value.as_slice())?;
-
-                if transfer_msg.to_address == query_data.recipient {
+                let recipient = query_data.iter().fold("", |prev, row| {
+                    if row.field == "transfer.address" && row.op.as_str() == "eq" {
```

Comment:
You can derive PartialEq and Eq traits with a `TransactionFilterOp` definition and compare row.op with a TransactionFilterOp::Eq not a str